### ISSUE TITLE
memory leak

### ISSUE THREAD

**Issue by amoodie** [amoodie/rivers2stratigraphy#1]
There seems to be a memory leak, whereby the process increases in memory size indefinitely with run time. This is after the channel bodies are removed from bottom.

**Comment by amoodie** [comment]
The memory leak is worse with the new PR #15 , I think I will need to dump the entire list periodically or try to explore why it gets longer and longer.

**Comment by amoodie** [comment]
https://mg.pov.lt/objgraph/

**Comment by amoodie** [comment]
resolved this issue, will open PR. Basically the issue was that new PatchCollection objects were being created during each iteration of the model, and not being garbage collected because they maintained reference (somehow? not sure?). The solution was to create and add the patch only once and then use the (largely undocumented) method of `PatchCollection.set_paths(patches)` to update the patches to draw, thereby only having one instance of the `PatchCollection`.

Documenting some useful debugging tools here:

```
from pympler.tracker import SummaryTracker, summary, muppy
tracker = SummaryTracker()
import types

objList = muppy.get_objects()
my_types = muppy.filter(objList, Type=(Channel, ChannelBody, PatchCollection, Polygon))
sum1 = summary.summarize(my_types)
summary.print_(sum1)
```

This prints the number of objects of a certain type, and can be called on each iteration, in order to track the number of objects being created.
